waltid-libraries/credentials/waltid-verification-policies/src/jvmMain/kotlin/id/walt/policies/policies/RevocationPolicy.jvm.kt

```diff
@@ -66,23 +67,25 @@ object Base64Utils {
 }
 
 object StreamUtils {
-  fun getBitValue(inputStream: InputStream, index: ULong, bitSize: Int) =
-    inputStream.bufferedReader().use { buffer ->
-      buffer.skip((index * bitSize.toULong()).toLong())
-      extractBitValue(buffer, index, bitSize.toULong())
+    fun getBitValue(inputStream: InputStream, index: ULong, bitSize: Int): List<Char> = inputStream.use { stream ->
+        //TODO: bitSize constraints
+        val bitStartPosition = index * bitSize.toUInt()
+        val byteStart = bitStartPosition / 8u
+        stream.skip(byteStart.toLong())
+        val bytesToRead = (bitSize - 1) / 8 + 1
+        extractBitValue(stream.readNBytes(bytesToRead), index, bitSize.toUInt())
```

Comment:
Perhaps consider defining some private constants for these values to improve clarity, e.g., BITS_PER_BYTE